lib/http/response/inflater.rb

```diff
@@ -16,7 +16,7 @@ def readpartial(*args)
         if chunk
           chunk = zstream.inflate(chunk)
         elsif !zstream.closed?
-          zstream.finish
+          zstream.finish if zstream.total_in > 0
```

Comment:
RuboCop suggests the following here:

```suggestion
          zstream.finish if zstream.total_in.positive?
```